src/cli.ts

```diff
@@ -61,15 +63,27 @@ async function run(argv: string[]) {
     return;
   }
 
-  const declarations = await generateDeclarations(args.root);
+  const fileMap = await generateDeclarations(args.root);
 
-  if (args.out) {
-    fs.writeFileSync(args.out, declarations);
+  if (args.outDir) {
+    console.log('Writing type declarations to ' + path.resolve(args.outDir));
+    await writeFileMap(args.outDir, fileMap);
   } else {
-    process.stdout.write(declarations);
+    const concatenated = [...fileMap.values()].join('\n');
+    process.stdout.write(concatenated);
   }
 }
 
+function writeFileMap(
+    rootDir: string, files: Map<string, string>): Promise<void[]> {
```

Comment:
It's a little weird to return a promise of an array of undefineds. Recommend instead returning just Promise<void> by awaiting the Promise.all rather than returning it.

Reply:
Done.